interface/patient_file/summary/demographics_print.php

```diff
@@ -6,56 +6,129 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
-// Currently this will print only a blank form, but some code was
-// preserved here and in options.inc.php to ease future support for
-// data for a specified patient.
+// This will print a blank form, and if "patientid" is specified then
+// any existing data for the specified patient is included.
 
 require_once("../../globals.php");
+
+// Option to substitute a custom version of this script.
+if (!empty($GLOBALS['gbl_rapid_workflow']) &&
+    $GLOBALS['gbl_rapid_workflow'] == 'LBFmsivd' &&
+    file_exists('../../../custom/demographics_print.php'))
+{
+  include('../../../custom/demographics_print.php');
+  exit();
+}
+
 require_once("$srcdir/acl.inc");
 require_once("$srcdir/options.inc.php");
 require_once("$srcdir/patient.inc");
 
+$patientid = empty($_REQUEST['patientid']) ? 0 : 0 + $_REQUEST['patientid'];
+if ($patientid < 0) $patientid = 0 + $pid; // -1 means current pid
+
+// True if to display as a form to complete, false to display as information.
+$isform = empty($_REQUEST['isform']) ? 0 : 1;
+
+// Html2pdf fails to generate checked checkboxes properly, so write plain HTML
+// if we are doing a patient-specific complete form.
+$PDF_OUTPUT = ($patientid && $isform) ? false : true;
+
+if ($PDF_OUTPUT) {
+  require_once("$srcdir/html2pdf/vendor/autoload.php");
+  $pdf = new HTML2PDF('P', 'Letter', 'en');
+  $pdf->setTestTdInOnePage(false); // Turn off error message for TD contents too big.
+  $pdf->pdf->SetDisplayMode('real');
+  ob_start();
+}
```

Comment:
For a more stable conversion from html/pdf that uses mPDF, check out @sjpadgett ' solution in the patient report.

Reply:
OK but later. :)